docs/products/opensearch/howto/opensearch-search-and-python.rst

```diff
@@ -0,0 +1,471 @@
+
+.. code-block:: python
+
+    client.search({
+        index: 'recipes',
+        q: 'ingredients:broccoli AND calories:(>=100 AND <200)'
+    })
+
+
+With Query DSL, the field ``body`` expects a dictionary object which makes it easier to construct more complex search queries. As an example:
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "multi_match": {
+                        "query": "Garlic-Lemon",
+                        "fields": [
+                          "title",
+                          "ingredients"
+                        ]
+                      }
+                    }
+                  }
+    resp = client.search(index=INDEX_NAME, body=query_body)
+    log_titles(resp)
+
+In this example, we are searching for "Garlic-Lemon" across ``title`` and ``ingredients`` fields.
+
+.. note::
+  In the method ``search()``. One of the optional fields is the ``size`` field, which is defined as the number of results returned in the search. The default value is 10.
+  
+
+We are not adjusting the ``size`` parameter in this tutorial. Therefore, we are using the default value which is 10 results per search.
+
+
+Common queries
+''''''''''''''
+
+In the next section, we will cover some of the more common queries. Let's start querying 🔎 
+
+Match query
+-----------
+
+The ``match`` query helps you to find the best matches with multiple search word and is the default option for full-text search. For example, if you want to find matches that in the ``title`` has "Chilled Tomato".
+This will return results of titles that contain "Chilled" or "Tomato" on it due to DSL defaults to the "or" operator.
+
+.. code-block:: python
+       query_body = {
+                      "query": {
+                        "match": {
+                          "title": "Chilled Tomato"
+                        }
+                      }
+                    }
+      resp = client.search(index=INDEX_NAME, body=query_body)
+      log_titles(resp)
+
+If you want to find exact matches for fields in the ``title`` as "Chilled Tomato", you can specify the operator "and".
+
+.. code-block:: python
+
+       query_body = {
+                      "query": {
+                        "match": {
+                          "title": {
+                            "query": "Chilled Tomato",
+                            "operator": "and"
+                          }
+                        }
+                      }
+                    }
+      resp = client.search(index=INDEX_NAME, body=query_body)
+      log_titles(resp)
+
+You can run yourself the code to explore the ``match`` function. For example, if you want to find spring recipes in the title:
+
+.. code-block:: shell
+
+  python search.py match "title" "Spring"
+  [
```

Comment:
To make it easier to copy the command, separate  the response from the example command line

Reply:
and cool that you add more examples how to run stuff in your demo project! I left another comment before saying that they are missing :D

Reply:
I actually tried to add more explanations on readme as well about the `service_uri`